src/components/Settings/AddGradeForm.tsx

```diff
@@ -0,0 +1,165 @@
+      studentId: '',
+      subject: '',
+      gradeId: '',
+      description: '',
+    },
+  });
+
+  const [selectedStudent, setSelectedStudent] = useState<SelectOption | null>(
+    null
+  );
+  const [selectedSubject, setSelectedSubject] = useState<SelectOption | null>(
+    null
+  );
+  const [selectedGrade, setSelectedGrade] = useState<SelectOption | null>(null);
+
+  const dispatch = useAppDispatch();
+  const { subjects } = useAppSelector((state) => state.subjects);
+  const { students } = useAppSelector((state) => state.students);
+  const { grades } = useAppSelector((state) => state.grades);
+
+  useEffect(() => {
+    dispatch(fetchSubjects());
+    dispatch(fetchStudents());
+    dispatch(fetchGrades());
+  }, [dispatch]);
+
+  const studentOptions: SelectOption[] =
+    students &&
+    Object.values(students).map((student: Student) => ({
+      value: student.id,
+      label: student.name,
+    }));
+
+  const subjectOptions: SelectOption[] =
+    subjects &&
+    Object.values(subjects).map((subject: Subject) => ({
+      value: subject.id,
+      label: subject.name,
+    }));
+  const gradeOptions: SelectOption[] =
+    grades &&
+    Object.values(grades).map((grade: Grades) => ({
+      value: grade.id,
+      label: grade.value,
+    }));
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    const newGrades = [
+      {
+        subjectId: data.subject,
+        grade: data.gradeId,
+        description: data.description,
+      },
+    ];
+
+    try {
+      const resultAction = await dispatch(
+        updateStudentWithGrades({
+          studentId: data.studentId,
+          newGrades,
+        })
+      );
+
+      if (updateStudentWithGrades.rejected.match(resultAction)) {
+        console.error('Failed to add grade:', resultAction.payload);
+      } else {
+        console.log('Grade added successfully:', resultAction.payload);
+        reset();
+        setSelectedStudent(null);
+        setSelectedSubject(null);
+        setSelectedGrade(null);
+      }
+    } catch (error) {
+      console.error('Error adding grade:', error);
+    }
+  };
+
+  return (
+    <form
+      className="max-w-sm mx-auto space-y-4"
+      onSubmit={handleSubmit(onSubmit)}
+    >
+      {/* Student select */}
+      <Select
+        options={studentOptions}
+        value={selectedStudent}
+        onChange={(selectedOption) => {
+          setSelectedStudent(selectedOption as SelectOption);
+          setValue('studentId', (selectedOption as SelectOption).value); // Set selected student ID
+        }}
+        placeholder="Select student..."
+        className="basic-select"
+        classNamePrefix="select"
+      />
+
+      {/* Subject select */}
+      <Select
+        options={subjectOptions}
+        value={selectedSubject}
+        onChange={(selectedOption) => {
+          setSelectedSubject(selectedOption as SelectOption);
+          setValue('subject', (selectedOption as SelectOption).value);
```

Comment:
Na názvy fieldu radeji použit nejakou konstatu. FIELD_NAMES: { SUBJECT: 'subject', GRADE: 'grade', .... }